### PR TITLE
Check structure of octokit errors instead of via prototype instance

### PR DESCRIPTION
**Pull request by dgellow**
That was a fun one.

A bit of context. As part of https://github.com/stainless-api/stainless/pull/2129 I am trying to debug an issue where `getBranchSha` throws an exception when it receives a GitHub 404 response. That was really surprising given the call to GitHub is wraps in a `try/catch` to handle that case:

```typescript
    try {
      const {
        data: {
          object: {sha},
        },
      } = await this.octokit.git.getRef({
        owner: this.repository.owner,
        repo: this.repository.repo,
        ref: `heads/${branchName}`,
      });
      this.logger.debug(`SHA for branch: ${sha}`);
      return sha;
    } catch (e) {
      // 👇 relevant code
      if (e instanceof RequestError && e.status === 404) {
        this.logger.debug(`Branch: ${branchName} does not exist`);
        return undefined;
      }
      throw e;
    }
```

By digging more into the generated JS and trying to debug it at runtime I can see that the constructor of the caught error is named "RequestError`, as expected, but the `instanceof` still returns `false` for both the imported `octokit.RequestError` and `@octokit/request-errors.RequestError`, pointing to a dependency mismatch.

By running `yarn list @octokit/request-error` I can see that between the project in Stainless repository and dependencies from `release-please` we have widely different versions of octokit packages. Unfortunately we cannot easily bump the octokit version used by `release-please`, octokit v3 requires node v18+ but `release-please` tests cannot be run with node v18+ (see https://github.com/stainless-api/release-please/pull/5).

So, long story short: this pull request uses type guards to structurally check octokit errors instead of relying on imported packages, side-stepping any dependency issue.

### CODE REVIEW

File: src/errors/index.ts [hunk 1]
```diff
@@ -91,3 +92,56 @@ export class FileNotFoundError extends Error {
...
+ * @param error The error object to check.
+ * @returns A boolean indicating whether the error is a RequestError.
+ */
+export function isOctokitRequestError(error: unknown): error is RequestError {
+  if (typeof error === 'object' && error !== null) {
+    const e = error as RequestError;
+    return (
+      e.name === 'HttpError' &&
+      typeof e.status === 'number' &&
+      typeof e.request === 'object'
```
Comment (RobertCraigie) [inline src/errors/index.ts:114]: question: is there anything that's more octokit-specific that we could check here as well?

Comment (dgellow) [reply src/errors/index.ts:114]: Not that I can see.

```ts
import { RequestOptions, ResponseHeaders, OctokitResponse } from "@octokit/types";
import { RequestErrorOptions } from "./types";
/**
 * Error with extra properties to help with debugging
 */
export declare class RequestError extends Error {
    name: "HttpError";
    /**
     * http status code
     */
    status: number;
    /**
     * http status code
     *
     * @deprecated `error.code` is deprecated in favor of `error.status`
     */
    code: number;
    /**
     * Request options that lead to the error.
     */
    request: RequestOptions;
    /**
     * error response headers
     *
     * @deprecated `error.headers` is deprecated in favor of `error.response.headers`
     */
    headers: ResponseHeaders;
    /**
     * Response object if a response was received
     */
    response?: OctokitResponse<unknown>;
    constructor(message: string, statusCode: number, options: RequestErrorOptions);
}

```

**Review by RobertCraigie — APPROVED** [review]
Gotta love JS!

File: src/errors/index.ts (hunk 1 above)
Comment (RobertCraigie) [reply src/errors/index.ts:114]: Good point, I don't think there's much chance of this causing issues anyway, seems like a fairly low surface area.

**Comment by dgellow** [comment]
Btw, I actually tested this branch by running Stainless API locally and updating the package.json. That fixes the problem with `trigger-release-please` :)

**Comment by RobertCraigie** [comment]
Ah amazing, thank you!
